holmes/plugins/toolsets/bash/validation.py

```
@@ -134,13 +267,201 @@ def parse_command_segments(command: str) -> Tuple[List[str], bool]:
+        return _unsafe_arg_result(
+            f"output redirection to '{target}' writes to the filesystem",
+            approval_mode,
+        )
+
+    # No hard deny. A runtime expansion ($(...), `...`, $VAR/${VAR}, <(...)) in an
+    # argv-checked command's arguments can expand into a blocked primitive that the
+    # static checks above cannot see, so require explicit approval rather than
+    # auto-allowing it.
+    for argv, arg_is_dynamic in zip(
+        extractor.command_argvs, extractor.command_arg_dynamic
+    ):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)